octobot_trading/exchanges/parser/positions_parser.py

```diff
@@ -0,0 +1,371 @@
+            [ExchangeCols.TIMESTAMP.value],
+            parse_method=parser_util.convert_any_time_to_seconds,
+            not_found_method=self.timestamp_not_found,
+        )
+
+    def mode_found(self, raw_mode):
+        if raw_mode in self.ONEWAY_VALUES:
+            return PositionMode.ONE_WAY
+        if raw_mode in self.HEDGE_VALUES:
+            return PositionMode.HEDGE
+        self._log_missing(
+            PositionCols.POSITION_MODE.value,
+            f"key: {self.MODE_KEY_NAMES}, "
+            f"got raw_mode: {raw_mode or 'no raw mode'}, "
+            f"allowed oneway values: {self.ONEWAY_VALUES}, "
+            f"allowed hedge values: {self.HEDGE_VALUES}",
+        )
+
+    def handle_missing_value(self, keys_to_find):
+        quantity = None
+        if (mark_price := self.formatted_record.get(PositionCols.MARK_PRICE.value)) and (
+                quantity := self.formatted_record.get(PositionCols.QUANTITY.value)
+        ):
+            return quantity / mark_price
+        self._log_missing(
+            PositionCols.VALUE.value,
+            f"keys: {keys_to_find} and using quantity ({quantity or 'no quantity'}) "
+            f"/ mark price ({mark_price or 'no mark price'})",
+        )
+
+    def quantity_found(self, quantity):
+        if quantity == constants.ZERO:
+            return quantity
+        if side := self.formatted_record.get(PositionCols.ORIGINAL_SIDE.value):
+            if side == PositionSide.LONG.value:
+                return quantity
+            elif side == PositionSide.SHORT.value:
+                # short - so we set it to negative if it isn't already
+                return -quantity if quantity > 0 else quantity
+        self._log_missing(
+            PositionCols.QUANTITY.value,
+            f"requires valid side ({side or 'no side'}) to parse",
+        )
+
+    def timestamp_not_found(self, _):
+        try:
+            return int(self.exchange.connector.get_exchange_current_time())
+        except Exception as e:
+            self._log_missing(
+                PositionCols.CONTRACT_TYPE.value,
+                f"Failed to get time with get_exchange_current_time: "
+                f"{self.formatted_record[PositionCols.SYMBOL.value]}",
+                error=e,
+            )
+
+
+# only keep keys here from exchanges that are 100% safe with any exchange
+class LiquidationSynonyms:
+    keys = ["bust_price"]
+
+
+class RealizedPnlSynonyms:
+    keys = [
+        "cum_realized_pnl",
+        "realized_pnl",
+        "cum_realised_pnl",
+        "realised_pnl",
+        "realisedPnl",
+    ]
+
+
+class UnrealizedPnlSynonyms:
+    keys = ["unrealizedPnl"]
+
+
+class ClosingFeeSynonyms:
+    keys = ["occ_closing_fee"]
+
+
+class ValueSynonyms:
+    keys = ["position_value"]
+
+
+class SizeSynonyms:
+    keys = [ExchangeCols.CONTRACTS.value, "positionAmt"]
+
+
+class QuantitySynonyms:
+    keys = [ExchangeCols.CONTRACT_SIZE.value]
+
+
+class SideSynonyms:
+    keys = ["positionSide"]
```

Comment:
Are we sure those keys are "100% safe with any exchange"?

Reply:
i would say yes, as the naming seems unique

Reply:
I don't understand why we should try to use bybit keys on other future exchanges.

Reply:
If just one other exchange uses the same naming, its already worth adding it. 
One key here, could make it or brake it for a exchange
No other way to support over 100 ccxt exchanges and potential hundreds of DEX and stock exchanges coming up

also there is no downside, keys are tested left to right, so if a exchange uses the default keys, it'll never try the synonyms. Only exchanges that would fail otherwise are even trying

Reply:
In my opinion, there is two things : 
- First we'll need to have generic parsers (without any ccxt code / keys / references)
- Secondly in ccxt parsers we should only found ccxt unified keys listed in https://docs.ccxt.com/en/latest/manual.html#unified-api

Reply:
With these two parser types we'll be able to support any unified ccxt exchange and also able to parse any other sources like cryptofeed

Reply:
yeah thats way better to split it up. I already refactored settings, trades, orders and positions parser. I'm continuing with the other parsers now